tests/providers/amazon/aws/hooks/test_base_aws.py

```diff
@@ -590,40 +590,40 @@ def import_mock(name, *args):
             [mock.call.get_default_id_token_credentials(target_audience="aws-federation.airflow.apache.org")]
         )
 
-    @mock.patch.object(
```

Comment:
Patching Connection at the test decorator level hits the DB while running test connection. What works is to move patching to inside the test